### PR TITLE
Fix disassociateDeadNodes usages

### PR DESCRIPTION
**Pull request by idegtiarenko**

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/coordination/JoinTaskExecutor.java [hunk 1]
```diff
@@ -236,7 +237,14 @@ protected ClusterState.Builder becomeMasterAndTrimConflictingNodes(
         logger.trace("becomeMasterAndTrimConflictingNodes: {}", tmpState.nodes());
         allocationService.cleanCaches();
         tmpState = PersistentTasksCustomMetadata.disassociateDeadNodes(tmpState);
-        return ClusterState.builder(allocationService.disassociateDeadNodes(tmpState, false, "removed dead nodes on election"));
+        return ClusterState.builder(
+            allocationService.disassociateDeadNodes(
+                tmpState,
+                false,
+                "removed dead nodes on election",
+                rerouteCompletionIsNotRequired() // reroute = false
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/cluster/coordination/JoinTaskExecutor.java:245]: Production code usage

File: server/src/main/java/org/elasticsearch/cluster/coordination/NodeRemovalClusterStateTaskExecutor.java [hunk 2]
```diff
@@ -74,7 +76,8 @@ public ClusterState execute(BatchExecutionContext<Task> batchExecutionContext) t
             finalState = allocationService.disassociateDeadNodes(
                 ptasksDisassociatedState,
                 true,
-                describeTasks(batchExecutionContext.taskContexts().stream().map(TaskContext::getTask).toList())
+                describeTasks(batchExecutionContext.taskContexts().stream().map(TaskContext::getTask).toList()),
+                rerouteCompletionIsNotRequired() // the change is not triggered by a user request
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/cluster/coordination/NodeRemovalClusterStateTaskExecutor.java:80]: Another production usage

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by idegtiarenko** [comment]
Closed in favor of https://github.com/elastic/elasticsearch/pull/89502

**Pull request closed without merging**